File: fadl_pos/fadl_pos/report/stock_value_by_warehouse/stock_value_by_warehouse.js
// Copyright (c) 2025, Ali Raza and contributors
// For license information, please see license.txt

frappe.query_reports["Stock Value by Warehouse"] = {
	filters: [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			reqd: 1,
			default: frappe.defaults.get_user_default("Company"),
			on_change: function () {
				frappe.query_report.set_filter_value("warehouse", null);
			},
		},
	],
};

File: fadl_pos/fadl_pos/report/stock_value_by_warehouse/stock_value_by_warehouse.py
# Copyright (c) 2026, FadlTech team and contributors

from __future__ import annotations



import frappe
from frappe import _


def execute(filters=None):
	columns, data = get_columns(), get_stock_value_by_warehouse(filters)
	return columns, data


def get_stock_value_by_warehouse(filters):
	query = """
        SELECT
            b.warehouse,
            SUM(b.stock_value) stock_value
        FROM `tabBin` b
        INNER JOIN `tabWarehouse` w ON b.warehouse = w.name
        WHERE
			b.actual_qty > 0
			AND b.warehouse IS NOT NULL
			AND b.warehouse != ''
			AND w.company = %(company)s
        GROUP BY b.warehouse
        HAVING SUM(b.stock_value) IS NOT NULL AND SUM(b.stock_value) > 0
    """
	return frappe.db.sql(query, {"company": filters.get("company")}, as_dict=True)


def get_columns():
	return [
		{
			"label": _("Warehouse"),
			"fieldname": "warehouse",
			"fieldtype": "Link",
			"options": "Warehouse",
			"width": 200,
		},
		{
			"label": _("Stock Value"),
			"fieldname": "stock_value",
			"fieldtype": "Currency",
			"align": "right",
			"width": 200,
		},
	]
